tests/phpunit/tests/block-supports/border.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * @group block-supports
+ */
+class Test_Block_Supports_Border extends WP_UnitTestCase {
+
+	function test_border_color_slug_with_numbers_is_kebab_cased_properly() {
+		$block_name = 'test/border-color-slug-with-numbers-is-kebab-cased-properly';
+		register_block_type(
+			$block_name,
+			array(
+				'api_version' => 2,
+				'attributes'  => array(
+					'borderColor' => array(
+						'type' => 'string',
+					),
+					'style'       => array(
+						'type' => 'object',
+					),
+				),
+				'supports'    => array(
+					'__experimentalBorder' => array(
+						'color'  => true,
+						'radius' => true,
+						'width'  => true,
+						'style'  => true,
+					),
+				),
+			)
+		);
+		$registry   = WP_Block_Type_Registry::get_instance();
+		$block_type = $registry->get_registered( $block_name );
+		$block_atts = array(
+			'borderColor' => 'red',
+			'style'       => array(
+				'border' => array(
+					'radius' => '10px',
+					'width'  => '1px',
+					'style'  => 'dashed',
+				),
+			),
+		);
+
+		$actual   = wp_apply_border_support( $block_type, $block_atts );
+		$expected = array(
+			'class' => 'has-border-color has-red-border-color',
+			'style' => 'border-radius: 10px; border-style: dashed; border-width: 1px;',
+		);
+
+		$this->assertSame( $expected, $actual );
+		unregister_block_type( $block_name );
```

Comment:
`$block_name` is unique for each test, so I suppose the refactor would be to assign the name string to `$this->block_name` for example in each test, and then call `unregister_block_type( $this->block_name )` in `tear_down()` (?) 

Personally I think it's okay for each test to explicitly unregister the block give the above.

Reply:
I think what @anton-vlasenko is proposing is more about ensuring that the blocks get unregistered when the assertion fails. In the current form, if the assertion would fail, then the block would remain registered.

Reply:
Oh that make sense. Apologies, I didn't consider that at all. I'll refactor.

Reply:
Threw up a PR https://github.com/WordPress/wordpress-develop/pull/2569

Thanks @anton-vlasenko and @gziolo 🙇